lib/send-mail.ts

```diff
@@ -60,15 +60,22 @@ export function parseMBox(mbox: string, gentle?: boolean):
     let subject: string | undefined;
     let to: string | undefined;
 
-    for (const line of header.split(/\n(?![ \t])/)) {
+    for (const line of header.split(/\n(?![ \t])/)
+        .map(e => e.replace("\r\n", " ").replace("\n", " "))) {
```

Comment:
Should this be done in `mail-archive-helper.ts`?  The 'other' user of this method does not generate `\r` in emails.  Is the split missing something that the second `replace`  is covering?  ie what use case are you seeing?

Reply:
IIRC, from the RFCs regarding SMTP mail, CRLF is the correct EOL mode for e-mails and I suspect that it's probably Git's EOL normalization behavior that is changing EOLs to LF only.
On the other hand, a LF without CR doesn't have a separate meaning, that's why from my POV:

- yes, it should be done here to handle both cases
- it is safe to do so

> Is the split missing something that the second replace is covering? ie what use case are you seeing?

One of the reasons why is dislike regex is that there's often some unexpected "twist" in the way it behaves, and this is one example where the answer is: Yes, the regex split is missing something

The intention of the regex is splitting of mail header lines while treating lines which are starting with a space or a tab as being part of the previous line.

The regex does that, and retains such consecutive as a single line in its output.
BUT: without removing the line breaks in those cases.

Please see the screenshot above where the message id is wrapped.

You can reproduce with an e-mail account from outlook.com

Reply:
Probably a little confusion here.  The other user is git generated mbox format emails which will only have LF characters.  These are the result of /submit and /preview commands.  Only `mail-archive-helper.ts` would be seeing SMTP conformant emails.

How does this work if the CRLF is in the messages being processed?  Could message content be processed as headers?

```
const headerEnd = mbox.indexOf("\n\n");
```

Please add test cases for the conditions you are encountering so that future changes will not accidentally ignore them.  The current tests do not handle the edge situations you are seeing.

Reply:
If this is staying, can it be:
```
.map(e => e.replace(/\r?\n\s+", " ").trim())) {
```
Would that remove the need for the later `replaceAll`?  No leading whitespace on any lines.

Reply:
I would prefer that form, too.

Reply:
How sure are you about the suggested code?
I don't want to spend the time to double check and verify that it behaves 100% correct in all possible cases.
Actually it was an unexpectedly behaving regex that caused the incorrect parsing in the first place.

Reply:
> How sure are you about the suggested code?

It was left as an exercise for the reader to figure out that the first double-quote should be a forward slash followed by a `g`, I think.

Reply:
LOL. You won't get me doing exercises with regex...

Let me explain. My experiences with regex had always been a pattern like this:

- Study the regex syntax
- Build a regex expression for the required situation
- Test it on a number of strings
- Implement code using the regex
- Publish code
- EACH TIME
  at some point there appeared a situation where the regex didn't do as expected
  => bug
- Needed to fix the bug
- Republish code
- Quickly afterwards: forget the regex syntax details

I had the same experience like 4 or 5 times in the past 20 years. Each time, something unexpected has happened.

On the other hand, I've written countless parsers for protocols or formats, bit-based or string-based. 
I write them once, I know exactly what they are doing, they are working rock-solid without surprises and I never need to touch it again (unless there are changes or extensions to the spec).

It goes without saying that there are regex experts that are capable of creating expressions that work without bad surprises, but I'm not such regex expert, and I prefer to write code in a way that I know it is solid.

Change my code, do it with regex, even close the PR and redo it yourself - all fine for me. It's surely a way to do things but not one of mine.  ;-)